forge/forge/models/config.py

```diff
@@ -20,16 +23,27 @@ def UserConfigurable(
     **kwargs,
 ) -> T:
     # TODO: use this to auto-generate docs for the application configuration
-    return Field(
+    field_info: FieldInfo = Field(
         default,
         *args,
         default_factory=default_factory,
-        from_env=from_env,
         description=description,
         exclude=exclude,
         **kwargs,
-        user_configurable=True,
     )
+    field_info.metadata.append("user_configurable")
```

Comment:
you *could* consider:
```suggestion
    field_info.metadata.append(("user_configurable", True))
```
this would allow `dict(field_info.metadata).get("user_configurable", False)`

Reply:
I changed to this and simplified `_get_field_metadata`